### PR TITLE
MODEL-1489: Allow marking Label with "is_benchmark_reference" flag

### PR DESCRIPTION
**Pull request by dubininsergey**
# Description

This PR enables marking import labels with "is_benchmark_reference" flag. Such labels will be set as benchmark references if the target project support "benchmark" quality mode.

https://www.loom.com/share/ccca6e59420b4848b5f146c9fdbe9950

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Document change (fix typo or modifying any markdown files, code comments or anything in the examples folder only)

## All Submissions

- [x] Have you followed the guidelines in our Contributing document?
- [x] Have you provided a description?
- [x] Are your changes properly formatted?

## New Feature Submissions

- [x] Does your submission pass tests?
- [x] Have you added thorough tests for your new feature?
- [ ] Have you commented your code, particularly in hard-to-understand areas?
- [ ] Have you added a Docstring?

## Changes to Core Features

- [x] Have you written new tests for your core changes, as applicable?
- [x] Have you successfully run tests with your changes locally?
- [ ] Have you updated any code comments, as applicable?

### CODE REVIEW

**Review by tomislav-peharda — DISMISSED** [review]

**Comment by dubininsergey** [comment]
The PR is awaiting for the complementary changes in the BE to be merged/deployed

**Review by tomislav-peharda — DISMISSED** [review]

File: libs/labelbox/src/labelbox/data/serialization/ndjson/converter.py [hunk 1]
```diff
@@ -106,14 +106,16 @@ def serialize(
...
 
-                res = example.dict(
+                res = annotation.dict(
                     by_alias=True,
                     exclude={"uuid"} if annotation_uuid == "None" else None,
                 )
                 for k, v in list(res.items()):
                     if k in IGNORE_IF_NONE and v is None:
                         del res[k]
+                if getattr(label, 'is_benchmark_reference'):
```
Comment (vbrodsky) [inline libs/labelbox/src/labelbox/data/serialization/ndjson/converter.py:119]: did you try to declare `class Label(_CamelCaseMixin)` , I do not think you would need this line

File: libs/labelbox/tests/data/annotation_import/test_data_types.py [hunk 2]
```diff
@@ -1,12 +1,8 @@
-import datetime
-from labelbox.schema.label import Label
 import pytest
-import uuid
 
 from labelbox.data.annotation_types.data import (
     AudioData,
     ConversationData,
-    DicomData,
```
Comment (vbrodsky) [inline libs/labelbox/tests/data/annotation_import/test_data_types.py:9]: why was this removed

File: libs/labelbox/tests/data/serialization/ndjson/test_rectangle.py [hunk 3]
```diff
@@ -85,3 +85,23 @@ def test_rectangle_mixed_start_end_points():
...
+    bbox = lb_types.ObjectAnnotation(
+        name="bbox",
+        value=lb_types.Rectangle(
+            start=lb_types.Point(x=81, y=28),
+            end=lb_types.Point(x=38, y=69),
+        ),
+        extra={"uuid": "c1be3a57-597e-48cb-8d8d-a852665f9e72"}
+    )
+
+    label = lb_types.Label(
```
Comment (vbrodsky) [inline libs/labelbox/tests/data/serialization/ndjson/test_rectangle.py:100]: can you test with `is_benchmark_reference` missing, to test the default value behavior

Comment (dubininsergey) [reply libs/labelbox/tests/data/serialization/ndjson/test_rectangle.py:100]: added! 🫡

File: libs/labelbox/tests/data/annotation_import/test_data_types.py (hunk 2 above)
Comment (dubininsergey) [reply libs/labelbox/tests/data/annotation_import/test_data_types.py:9]: the imports were unused

File: libs/labelbox/src/labelbox/data/serialization/ndjson/converter.py (hunk 1 above)
Comment (dubininsergey) [reply libs/labelbox/src/labelbox/data/serialization/ndjson/converter.py:119]: I tend to avoid juggling with mixins because this caused MRO problems in the past. Do you think we can leave it as is?

Comment (vbrodsky) [reply libs/labelbox/src/labelbox/data/serialization/ndjson/converter.py:119]: the way I have it above should prob work, can you try? If it does not work (i.e. test fails etc) we could use your approach, but I rather not have custom serialization if we can avoid it

Comment (dubininsergey) [reply libs/labelbox/src/labelbox/data/serialization/ndjson/converter.py:119]: actually, the logic is the following: if there is `is_benchmark_reference` attribute set for the label, then all child annotations should have `isBenchmarkReferenceLabel` set to True. I don't see how the code above would help to avoid this logic

**Review by vbrodsky — APPROVED** [review]
